A3A/addons/core/functions/AI/fn_fastrope.sqf

```diff
@@ -19,57 +73,182 @@ private _wp = _heli addWaypoint [_landpos, 0];
 _wp setWaypointType "MOVE";
 _wp setWaypointBehaviour "CARELESS";
 _wp setWaypointSpeed "FULL";
+
 _wp setWaypointCompletionRadius 3;
 
+waitUntil {sleep 1; (not alive _veh) or (_veh distance _landpos < 550) or !(canMove _veh)};
+private _midHeight = [100, 150] select (A3A_climate isEqualTo "tropical");
+if (_vehType in FactionGet(all,"vehiclesPlanesTransport")) then {
+	_veh flyInHeight _midHeight;
+}else{};
+// Landing path setup for vtol
+private _endPos = _landpos;
+private _startPos = getPosASL _veh;
+private _midPos = _endPos vectorAdd [0,0,_midHeight];
+private _initialVelocity = (velocity _veh);
+_initialVelocity set [2, 0];
+private _velocityVector = +_initialVelocity;
+_initialVelocity = vectorMagnitude _initialVelocity;
+private _initialSpeed = speed _veh/3.6;
+//We got the initial velocity of the heli
+private _distance = _startPos distance _midPos;
+private _landingTime = _distance/_initialVelocity * 1.35;
+private _maxAngle = ((_initialVelocity * _initialVelocity/3600) * 35) min 35;
+//Starting land approach with bezier curve
+private _startToMidVector = _midPos vectorDiff _startPos;
+private _midToEndVector = _endPos vectorDiff _midPos;
+private _vectorDir = vectorDir _veh;
+private _vectorUp = vectorUp _veh;
+private _interval = 0;
+private _time = 0;
+private _angleStep = 0.25;
+private _angleTarget = 0;
+private _angleIs = 0;
+private _angleDiff = 0;
+private _heightDiff = 0;
+private _driver = driver _veh;
```

Comment:
Is all of this for vtols specifically?

Reply:
Yep, part of combat landing script, otherwise VTOL will overshoot and never get to waypoint

Reply:
does it need to be grabbed even if the vehicle is not a VTOL?

Reply:
I might add a check for VTOL but wouldn't it make all the data in if statement inaccessible outside it?

Reply:
Depends on how you do it

Reply:
Haven't tried adding a check, but I'm really sceptical about it, maybe I should wright a separate functions for VTOL and maybe combat capable helis?

Reply:
perhaps